network/alsp/manager/manager_test.go

```diff
@@ -188,6 +188,94 @@ func TestHandleReportedMisbehavior_Integration(t *testing.T) {
+// The test ensures that despite attempting on connections, no inbound or outbound connections between the victim and
+// the disallow-listed spammer node are established.
+func TestHandleReportedMisbehavior_And_DisallowListing_Integration(t *testing.T) {
+	cfg := managerCfgFixture()
+
+	// create a new MisbehaviorReportManager
+	var victimSpamRecordCacheCache alsp.SpamRecordCache
+	cfg.Opts = []alspmgr.MisbehaviorReportManagerOption{
+		alspmgr.WithSpamRecordsCacheFactory(func(logger zerolog.Logger, size uint32, metrics module.HeroCacheMetrics) alsp.SpamRecordCache {
+			victimSpamRecordCacheCache = internal.NewSpamRecordCache(size, logger, metrics, model.SpamRecordFactory())
+			return victimSpamRecordCacheCache
+		}),
+	}
+
+	ids, nodes, mws, _, _ := testutils.GenerateIDsAndMiddlewares(
+		t,
+		3,
+		unittest.Logger(),
+		unittest.NetworkCodec(),
+		unittest.NetworkSlashingViolationsConsumer(unittest.Logger(), metrics.NewNoopCollector()))
+	sms := testutils.GenerateSubscriptionManagers(t, mws)
+	networkCfg := testutils.NetworkConfigFixture(t, unittest.Logger(), *ids[0], ids, mws[0], sms[0], p2p.WithAlspConfig(cfg))
+	victimNetwork, err := p2p.NewNetwork(networkCfg)
+	require.NoError(t, err)
+
+	// index of the victim node in the nodes slice.
+	victimIndex := 0
+	// index of the spammer node in the nodes slice (the node that will be reported for misbehavior and disallow-listed by victim).
+	spammerIndex := 1
+	// other node (not victim and not spammer) that we have to ensure is not affected by the disallow-listing of the spammer.
+	honestIndex := 2
+
+	ctx, cancel := context.WithCancel(context.Background())
+	signalerCtx := irrecoverable.NewMockSignalerContext(t, ctx)
+	testutils.StartNodesAndNetworks(signalerCtx, t, nodes, []network.Network{victimNetwork}, 100*time.Millisecond)
+	defer testutils.StopComponents[p2p.LibP2PNode](t, nodes, 100*time.Millisecond)
+	defer cancel()
+
+	p2ptest.LetNodesDiscoverEachOther(t, ctx, nodes, ids)
+	// initially victim and spammer should be able to connect to each other.
+	p2ptest.TryConnectionAndEnsureConnected(t, ctx, nodes)
+
+	e := mocknetwork.NewEngine(t)
+	con, err := victimNetwork.Register(channels.TestNetworkChannel, e)
+	require.NoError(t, err)
+
+	// creates a misbehavior report for the spammer
+	report := misbehaviorReportFixtureWithPenalty(t, ids[spammerIndex].NodeID, model.DefaultPenaltyValue)
+
+	// imitates that the victim has detected the spammer on 120 times of violations and reports the misbehavior
+	// to the network. As each report has the default penalty, ideally the spammer should be disallow-listed after
+	// 100 reports (each having 0.01 * disallow-listing penalty). But we take 120 as a safe number to ensure that
+	// the spammer is disallow-listed definitely.
+	reportCount := 120
+	wg := sync.WaitGroup{}
+	for i := 0; i < reportCount; i++ {
+		wg.Add(1)
+		// reports the misbehavior
+		report := report // capture range variable
```

Comment:
There's a shadow variable warning if both variables are named the same.
```suggestion
		wgReport := report // capture range variable
```

Reply:
resolved all shadow variables in https://github.com/onflow/flow-go/pull/4441/commits/37bedd3d691e5fb0ebed609bfcaebf2979171e93